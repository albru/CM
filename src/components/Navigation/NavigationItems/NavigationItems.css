.NavigationItems {
    margin: 0;
    padding: 0;
    list-style: none;
    display: flex;
    flex-flow: column;
    justify-content: center;
    height: 100%;
}

.NavToolbar {
    align-items: flex-start;
}

.NavFooter {
    align-items: center;
}

@media (min-width: 500px) {
    .NavigationItems {
        flex-flow: row;
        align-items: center;
        margin-bottom: 30px;
    }
}

