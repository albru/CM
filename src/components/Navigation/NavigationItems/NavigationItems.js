import React from 'react';
import PropTypes from 'prop-types';

import NavigationItem from './NavigationItem/NavigationItem';

import classes from './NavigationItems.css';

const NavigationItems = props => {
    return (
        <ul className={[classes.NavigationItems, classes[props.navType]].join(' ')}>
            <NavigationItem route={'/services'}>
                Услуги
            </NavigationItem>

            <NavigationItem route={'/portfolio'}>
                Портфолио
            </NavigationItem>

            <NavigationItem route={'/order'}>
                Размещение рекламы
            </NavigationItem>

            <NavigationItem route={'/contacts'}>
                Контакты
            </NavigationItem>

            {props.auth ? <NavigationItem route={'/orders'}>Мои заказы</NavigationItem> : null}
            
            {props.auth 
                ? <NavigationItem route={'/logout'}>Выйти</NavigationItem>
                : <NavigationItem route={'/auth'}>Войти</NavigationItem>}
        </ul>
    );
};

NavigationItem.propTypes = {
    navType: PropTypes.string
}

export default NavigationItems;
