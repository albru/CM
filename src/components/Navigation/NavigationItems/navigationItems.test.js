import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import NavigationItems from './NavigationItems';

configure({ adapter: new Adapter() })

describe('Modal', () => {
    it('should render correctly', () => {
        const output = shallow(
            <NavigationItems />
        );
        expect(shallowToJson(output)).toMatchSnapshot();
    });
});