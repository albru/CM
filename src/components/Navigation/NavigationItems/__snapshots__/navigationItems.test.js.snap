// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationItems should render correctly 1`] = `
<ul
  className=" "
>
  <NavigationItem
    key="Услуги"
    name="Услуги"
    route="/services"
  />
  <NavigationItem
    key="Портфолио"
    name="Портфолио"
    route="/portfolio"
  />
  <NavigationItem
    key="Конструктор рекламы"
    name="Конструктор рекламы"
    route="/constructor"
  />
  <NavigationItem
    key="Контакты"
    name="Контакты"
    route="/contacts"
  />
</ul>
`;
